tests/framework/extensions/clusters/clusters.go

```diff
@@ -163,7 +163,7 @@ func NewRKE1ClusterConfig(clusterName, cni, kubernetesVersion string, psact stri
 }
 
 // NewK3SRKE2ClusterConfig is a constructor for a apisV1.Cluster object, to be used by the rancher.Client.Provisioning client.
-func NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredentialSecretName, kubernetesVersion string, psact string, machinePools []apisV1.RKEMachinePool, advancedOptions rancherProvisioning.AdvancedOptions) *apisV1.Cluster {
+func NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredentialSecretName, kubernetesVersion string, psact string, machinePools []apisV1.RKEMachinePool, advancedOptions rancherProvisioning.AdvancedOptions, s3Snapshot *rkev1.ETCDSnapshotS3) *apisV1.Cluster {
```

Comment:
Instead of adding a new parameter to the ClusterConfig and you use the TestRKE2Cluster can you do this instead? https://github.com/rancher/rancher/blob/c2423288d53210b4f1526e22c9e9ff18de1b24a8/tests/v2/validation/provisioning/rke2/ace_test.go#L148-L150 where you add the snap shot after `NewK3SRKE2ClusterConfig` less overhead.

Reply:
I agree with this suggestion, that way we keep the `etcd` related things in the `etcd` tests while not necessarily changing this function and related files